### ISSUE TITLE
Variable substitution doesn't work across files

### ISSUE THREAD

**Issue by jcechace** [bkeepers/dotenv#312]
Variable substitution evaluates to values defined in the same file first. 
If the referenced variable is present in the same file, this "local" value will take precedence over exported system variable -- even existing environment variables will be ignore.

### Steps to reproduce
```
require 'dotenv'
Dotenv.load('env_custom', 'env_default')
puts ENV['STATEMENT']
```
Paste your env using:
```bash
# env_default
WHO="John"
STATEMENT="${WHO} knows nothing!"
```
```bash
# env_custom
WHO="Ygritte"
```

### Expected behavior
```"Ygritte knows nothing"```

### Actual behavior
``` "John knows nothing"```

### System configuration
**dotenv version**:  2.2
**Ruby version**: 2.4.0

**Comment by jcechace** [comment]
The following would work as expected

```ruby
require 'dotenv'
Dotenv.load('env_third', 'env_custom', 'env_default')
puts ['STATEMENT']
```

```bash
# env_default -no WHO variable defined in this file
STATEMENT="${WHO} knows nothing!"
```
```bash
# env_custom
WHO="Ygritte"
```
```bash
# env_third
WHO="Arya"
```

### Expected behavior
```"Arya knows nothing"```

**Comment by jcechace** [comment]
The issue is here https://github.com/bkeepers/dotenv/blob/master/lib/dotenv/substitutions/variable.rb#L28

I can send a pull request if you would accept one.

**Comment by cadwallion** [comment]
This has been resolved by #313
